File: JAVA/Cola/PrincipalCola.java
package JAVA.Cola;

import java.util.InputMismatchException;
import java.util.Scanner;

public class PrincipalCola {
    public static void main(String[] args) {
        Cola cola = new Cola();
        Scanner key = new Scanner(System.in);

        int op = 0;
        do {
            System.out.println("******MENU ESTUDIANTE*******");
            System.out.println("1. Agregar");
            System.out.println("2. Buscar");
            System.out.println("3. Ordenar");
            System.out.println("4. Desordenar");
            System.out.println("5. Eliminar");
            System.out.println("6. Insertar");
            System.out.println("7. Recorrer coleccion de INICIO A FIN");
            System.out.println("8. Recorrer coleccion de FIN A INICIO");
            System.out.println("9. Imprimir colección");
            System.out.println("10. Salir");
            System.out.println("******MENU ESTUDIANTE*******");

            try {
                System.out.print(">");
                op = key.nextInt();
            } catch (InputMismatchException e) {
                System.out.println("****************************");
                System.out.println("\tOpcion invalida");
                System.out.println("****************************");
                key = new Scanner(System.in);
            }

            key = new Scanner(System.in);
            switch (op) {
                case 1:
                    key = new Scanner(System.in);
                    System.out.println("**********AGREGAR**********");
                    System.out.print("Ingrese los nombres del Estudiante: ");
                    String nombres = key.nextLine();

                    System.out.print("Ingrese los apellidos del Estudiante: ");
                    String apellidos = key.nextLine();

                    System.out.print("Ingrese el tipo de identificacion del Estudiante: ");
                    String tipoIdentificacion = key.next();

                    System.out.print("Ingrese el numero de identificacion del Estudiante: ");
                    int numeroIdentificacion = key.nextInt();

                    System.out.print("Ingrese el correo electrónico del Estudiante: ");
                    String correoElectronico = key.next();

                    System.out.print("Ingrese el teléfono del Estudiante: ");
                    String telefono = key.next();

                    key = new Scanner(System.in);
                    System.out.print("Ingrese la dirección del Estudiante: ");
                    String direccion = key.nextLine();

                    System.out.print("Ingrese la fecha de nacimiento del Estudiante: ");
                    String fechaNacimiento = key.next();

                    System.out.print("Ingrese la fecha de ingreso del Estudiante: ");
                    String fechaIngreso = key.next();

                    System.out.print("Ingrese la fecha de egreso del Estudiante: ");
                    String fechaEgreso = key.next();

                    key = new Scanner(System.in);
                    System.out.print("Ingrese el estado del Estudiante: ");
                    String estado = key.nextLine();

                    System.out.print("Ingrese el semestre que cursa el Estudiante: ");
                    int semestre = key.nextInt();

                    key = new Scanner(System.in);
                    System.out.print("Ingrese la carrera que cursa el Estudiante: ");
                    String carrera = key.nextLine();

                    Estudiante nuevo = new Estudiante(nombres, apellidos, tipoIdentificacion, numeroIdentificacion,
                            correoElectronico, telefono, direccion, fechaNacimiento, fechaIngreso, fechaEgreso, estado,
                            semestre, carrera);
                    cola.agregar(nuevo);
                    key = new Scanner(System.in);
                    System.out.println("\nESTUDIANTE AGREGADO EXITOSAMENTE");
                    System.out.println("PRESIONE UNA TECLA PARA ENTER/INTRO PARA IR AL MENU...");
                    key.nextLine();
                    System.out.println("**********AGREGAR**********\n");
                    break;

                case 2:
                    System.out.println("**********BUSCAR**********");
                    System.out.print("Ingrese el numero de Identificacion del Estudiante: ");
                    int busqueda = key.nextInt();

                    Estudiante respuesta = cola.buscar(busqueda);
                    
                    if(respuesta != null){
                        System.out.println("El estudiante "+respuesta.getNombres()+" "
                        + respuesta.getApellidos()+" con el numero de identificacion "+respuesta.getNumeroIdentificacion()+" esta en la cola");
                    }else{
                        System.out.println("El estudiante no se encuentra en la cola");
                    }

                    System.out.println("PRESIONE UNA TECLA PARA ENTER/INTRO PARA IR AL MENU...");
                    key.nextLine();
                    System.out.println("**********BUSCAR**********");
                    break;

                case 3:
                    System.out.println("**********ORDENAR**********");
                    cola.ordenar();
                    System.out.println("COLA ORDENADA POR NUMERO DE IDENTIFICACION");
                    System.out.println("PRESIONE UNA TECLA PARA ENTER/INTRO PARA IR AL MENU...");
                    key.nextLine();
                    System.out.println("**********ORDENAR**********");
                    break;

                case 4:
                    System.out.println("**********DESORDENAR**********");
                    cola.desordenar();
                    System.out.println("COLA DESORDENADA POR NUMERO DE IDENTIFICACION");
                    System.out.print("PRESIONE UNA TECLA PARA ENTER/INTRO PARA IR AL MENU...");
                    key.nextLine();
                    System.out.println("**********DESORDENAR**********");
                    break;

                case 5:
                    System.out.println("**********ELIMINAR**********");
                    System.out.print("Ingrese el numero de identificacion del Estudiante a eliminar: ");
                    int eliminar = key.nextInt();
                    
                    if(cola.eliminar(eliminar)){
                        System.out.println("El estudiante con el numero de identificacion "+eliminar+" ha sido eliminado");
                    }else {
                        System.out.println("El estudiante con el numero de identificacion "+eliminar+" no se encuentra en la cola");
                    }
                    
                    System.out.println("PRESIONE UNA TECLA PARA ENTER/INTRO PARA IR AL MENU...");
                    key.nextLine();
                    System.out.println("**********ELIMINAR**********");
                    break;

                case 6:
                    key = new Scanner(System.in);
                    System.out.println("**********INSERTAR**********");
                    System.out.print("Ingrese los nombres del Estudiante: ");
                    String nombresIn = key.nextLine();

                    System.out.print("Ingrese los apellidos del Estudiante: ");
                    String apellidosIn = key.nextLine();

                    System.out.print("Ingrese el tipo de identificacion del Estudiante: ");
                    String tipoIdentificacionIn = key.next();

                    System.out.print("Ingrese el numero de identificacion del Estudiante: ");
                    int numeroIdentificacionIn = key.nextInt();

                    System.out.print("Ingrese el correo electrónico del Estudiante: ");
                    String correoElectronicoIn = key.next();

                    System.out.print("Ingrese el teléfono del Estudiante: ");
                    String telefonoIn = key.next();

                    key = new Scanner(System.in);
                    System.out.print("Ingrese la dirección del Estudiante: ");
                    String direccionIn = key.nextLine();

                    System.out.print("Ingrese la fecha de nacimiento del Estudiante: ");
                    String fechaNacimientoIn = key.next();

                    System.out.print("Ingrese la fecha de ingreso del Estudiante: ");
                    String fechaIngresoIn = key.next();

                    System.out.print("Ingrese la fecha de egreso del Estudiante: ");
                    String fechaEgresoIn = key.next();

                    key = new Scanner(System.in);
                    System.out.print("Ingrese el estado del Estudiante: ");
                    String estadoIn = key.nextLine();

                    System.out.print("Ingrese el semestre que cursa el Estudiante: ");
                    int semestreIn = key.nextInt();

                    key = new Scanner(System.in);
                    System.out.print("Ingrese la carrera que cursa el Estudiante: ");
                    String carreraIn = key.nextLine();

                    Estudiante nuevoIn = new Estudiante(nombresIn, apellidosIn, tipoIdentificacionIn, numeroIdentificacionIn,
                            correoElectronicoIn, telefonoIn, direccionIn, fechaNacimientoIn, fechaIngresoIn, fechaEgresoIn, estadoIn,
                            semestreIn, carreraIn);

                    int pos = 0;
                    do{
                        key = new Scanner(System.in);
                        System.out.print("Ingrese la posición donde desea insertar el Estudiante: ");
                        pos = key.nextInt();
                        
                        if(pos > cola.getSize()){
                            System.out.println("La posición ingresada es mayor a la cantidad de estudiantes en la cola");
                        }
                    }while(pos < 0 || pos > cola.getSize());
                    
                    cola.insertar(pos, nuevoIn);
                    
                    key = new Scanner(System.in);
                    System.out.println("ESTUDIANTE INSERTADO EN LA POSICION ["+pos+"]");
                    System.out.print("PRESIONE UNA TECLA PARA ENTER/INTRO PARA IR AL MENU...");
                    key.nextLine();
                    System.out.println("**********INSERTAR**********");
                    break;

                case 7:
                    System.out.println("**********RECORRER DE INICIO A FIN**********");
                    cola.recorrerInicioFin();
                    
                    key = new Scanner(System.in);
                    System.out.print("PRESIONE UNA TECLA PARA ENTER/INTRO PARA IR AL MENU...");
                    key.nextLine();
                    System.out.println("**********RECORRER DE INICIO A FIN**********");
                    break;

                case 8:
                    System.out.println("**********RECORRER DE FIN A INICIO**********");
                    cola.recorrerFinInicio();

                    key = new Scanner(System.in);
                    System.out.print("PRESIONE UNA TECLA PARA ENTER/INTRO PARA IR AL MENU...");
                    key.nextLine();
                    System.out.println("**********RECORRER DE FIN A INICIO**********");
                    break;

                case 9:
                    System.out.println("**********IMPRIMIR**********");
                    cola.imprimir();
                    key = new Scanner(System.in);
                    System.out.println("PRESIONE UNA TECLA PARA ENTER/INTRO PARA IR AL MENU...");
                    key.nextLine();
                    System.out.println("**********IMPRIMIR**********\n");
                    break;
                default:
                    break;
            }
        } while (op != 10);
    }
}

File: PYTHON/Cola/PrincipalCola.py
from Cola import Cola
from Estudiante import Estudiante

cola = Cola()

ctrl = True
while ctrl:
    opc = 0
    print("******MENU ESTUDIANTE*******")
    print("1. Agregar")
    print("2. Buscar")
    print("3. Ordenar")
    print("4. Desordenar")
    print("5. Eliminar")
    print("6. Insertar")
    print("7. Recorrer IMPRIMIR de INICIO A FIN")
    print("8. Recorrer IMPRIMIR de FIN A INICIO")
    print("9. Imprimir colección")
    print("10. Salir")
    print("******MENU ESTUDIANTE*******")
    
    try:
        opc = int(input("Ingrese una opcion: "))
    except ValueError:
        print("******ERROR*******")
        print("Opcion invalida")
        print("******ERROR*******")
        input("PRESIONE UNA TECLA PARA ENTER/INTRO PARA IR AL MENU...")

    if opc == 1:
        print("**********AGREGAR**********")
        nombres = input("Ingrese los nombres del Estudiante: ")
        apellidos = input("Ingrese los apellidos del Estudiante: ")
        tipoIdentificacion = input("Ingrese el tipo de identificacion del Estudiante: ")
        numeroIdentificacion = int(input("Ingrese el numero de identificacion del Estudiante: "))
        email = input("Ingrese el correo electronico del Estudiante: ")
        telefono = input("Ingrese el telefono del Estudiante: ")
        direccion = input("Ingrese el direccion del Estudiante: ")
        fechaNacimiento = input("Ingrese la fecha de nacimiento del Estudiante: ")
        fechaIngreso = input("Ingrese la fecha de ingreso del Estudiante: ")
        fechaEgreso = input("Ingrese la fecha de egreso del Estudiante: ")
        estado = input("Ingrese el estado del Estudiante: ")
        semestre = int(input("Ingrese el semestre del Estudiante: "))
        carrera = input("Ingrese la carrera del Estudiante: ")

        nuevo = Estudiante(nombres, apellidos, tipoIdentificacion, numeroIdentificacion, email, telefono, direccion, fechaNacimiento, fechaIngreso, fechaEgreso, estado, semestre, carrera)
        cola.agregar(nuevo)
        print("ESTUDIANTE AGREGADO A LA COLA")
        input("PRESIONE UNA TECLA PARA ENTER/INTRO PARA IR AL MENU...")
        print("**********AGREGAR**********")

    elif opc == 2:
        print("**********BUSCAR**********")
        numeroIdentificacion = int(input("Ingrese el numero de identificacion del Estudiante a buscar: "))

        if numeroIdentificacion == None:
            print("NO HA INGRESADO NINGUNA CEDULA")
        
        else:
            busqueda = cola.buscar(numeroIdentificacion)
            if busqueda == None:
                print("NO HAY ESTUDIANTES CON ESTA CEDULA")
            else:
                print("El Estudiante de numero de identificación "+str(numeroIdentificacion)+" "+busqueda.getNombres()+" "+busqueda.getApellidos()+" está en la cola")
            input("PRESIONE UNA TECLA PARA ENTER/INTRO PARA IR AL MENU...")
            print("**********BUSCAR**********")
        
    elif opc == 3:
        print("**********ORDENAR**********")
        cola.ordenar()
        print("COLA ORDENADA POR NUMERO DE IDENTIFICACION DE MENOR A MAYOR")
        input("PRESIONE UNA TECLA PARA ENTER/INTRO PARA IR AL MENU...")
        print("**********ORDENAR**********")

    elif opc == 4:
        print("**********DESORDENAR**********")
        cola.desordenar()
        print("COLA DESORDENADA")
        input("PRESIONE UNA TECLA PARA ENTER/INTRO PARA IR AL MENU...")
        print("**********DESORDENAR**********")

    elif opc == 5:
        print("**********ELIMINAR**********")
        numeroIdentificacion = int(input("Ingrese el numero de identificacion del Estudiante a eliminar: "))
        
        if numeroIdentificacion == None:
            print("NO HA INGRESADO NINGUNA CEDULA")
        
        else:
            busqueda = cola.buscar(numeroIdentificacion)
            if busqueda == None:
                print("NO HAY ESTUDIANTES CON ESTA CEDULA")
            else:
                cola.eliminar(numeroIdentificacion)
                print("EL ESTUDIANTE DE NUMERO DE IDENTIFICACION "+str(numeroIdentificacion)+" "+busqueda.getNombres()+" "+busqueda.getApellidos()+" HA SIDO ELIMINADO")
        input("PRESIONE UNA TECLA PARA ENTER/INTRO PARA IR AL MENU...")
        print("**********ELIMINAR**********")

    elif opc == 6:
        print("**********INSERTAR**********")
        nombres = input("Ingrese los nombres del Estudiante: ")
        apellidos = input("Ingrese los apellidos del Estudiante: ")
        tipoIdentificacion = input("Ingrese el tipo de identificacion del Estudiante: ")
        numeroIdentificacion = int(input("Ingrese el numero de identificacion del Estudiante: "))
        email = input("Ingrese el correo electronico del Estudiante: ")
        telefono = input("Ingrese el telefono del Estudiante: ")
        direccion = input("Ingrese el direccion del Estudiante: ")
        fechaNacimiento = input("Ingrese la fecha de nacimiento del Estudiante: ")
        fechaIngreso = input("Ingrese la fecha de ingreso del Estudiante: ")
        fechaEgreso = input("Ingrese la fecha de egreso del Estudiante: ")
        estado = input("Ingrese el estado del Estudiante: ")
        semestre = int(input("Ingrese el semestre del Estudiante: "))
        carrera = input("Ingrese la carrera del Estudiante: ")
        pos = int(input("Ingrese la posición donde desea insertar el Estudiante: "))

        nuevo = Estudiante(nombres, apellidos, tipoIdentificacion, numeroIdentificacion, email, telefono, direccion, fechaNacimiento, fechaIngreso, fechaEgreso, estado, semestre, carrera)
        
        if(pos > 0 and pos <= cola.size):
            cola.insertar(pos, nuevo)
            print("ESTUDIANTE INSERTADO EN LA POSICION ",pos)
        elif(pos == 0):
            print("LA COLECCION COMIENZA EN 1")
        else:
            print("Posición invalida")
            input("PRESIONE UNA TECLA PARA ENTER/INTRO PARA IR AL MENU...")
            print("**********INSERTAR**********")
    
    elif opc == 7:
        print("**********RECORRER DE INICIO A FIN**********")
        cola.recorrerInicioFin()
        input("PRESIONE UNA TECLA PARA ENTER/INTRO PARA IR AL MENU...")
        print("**********RECORRER DE INICIO A FIN**********")

    elif opc == 8:
        print("**********RECORRER DE FIN A INICIO**********")
        cola.recorrerFinInicio()
        input("PRESIONE UNA TECLA PARA ENTER/INTRO PARA IR AL MENU...")
        print("**********RECORRER DE FIN A INICIO**********")

    elif opc == 9:
        print("**********IMPRIMIR**********")
        cola.imprimir()
        print("**********IMPRIMIR**********")
        input("PRESIONE UNA TECLA PARA ENTER/INTRO PARA IR AL MENU...")
        print("**********IMPRIMIR**********")

    elif opc == 10:
        ctrl = False